libraries/stdlib/src/kotlin/collections/MapWithDefault.kt

```diff
@@ -45,6 +48,9 @@ public fun <K, V> Map<K, V>.withDefault(defaultValue: (key: K) -> V): Map<K, V>
  * and a value is obtained with [Map.getValue] function, for example when properties are delegated to the map.
  *
  * When this map already has an implicit default value provided with a former call to [withDefault], it is being replaced by this call.
+ *
+ * @sample samples.collections.Maps.Usage.getValueWithDefault
+ * @sample samples.collections.Maps.Usage.getValueWithReplacedDefault
```

Comment:
Makes sense to add another sample specific for MutableMaps with default, showing that the changes made to the returned map with a default are propagated back to the original mutable map.

Reply:
added `changesToMutableMapWithDefaultPropagateToUnderlyingMap`